packages/web-api-scan-runner/src/tasks/notification-dispatcher.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Queue, StorageConfig } from 'azure-services';
+import { ScanRunTimeConfig, ServiceConfiguration, System } from 'common';
+import { inject, injectable } from 'inversify';
+import { Logger, loggerTypes } from 'logger';
+import { OnDemandPageScanRunResultProvider } from 'service-library';
+import {
+    NotificationError,
+    NotificationState,
+    OnDemandNotificationRequestMessage,
+    OnDemandPageScanResult,
+    ScanCompletedNotification,
+} from 'storage-documents';
+
+// tslint:disable: no-null-keyword no-any
+
+@injectable()
+export class NotificationDispatcher {
+    constructor(
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
+        @inject(StorageConfig) private readonly storageConfig: StorageConfig,
+        @inject(Queue) private readonly queue: Queue,
+        @inject(Logger) private readonly logger: Logger,
+        @inject(loggerTypes.Process) private readonly currentProcess: typeof process,
+        private readonly system: typeof System = System,
+    ) {}
+
+    public async dispatchOnDemandScanRequests(onDemandNotificationRequestMessage: OnDemandNotificationRequestMessage): Promise<void> {
+        this.logger.logInfo(`Reading page scan run result ${onDemandNotificationRequestMessage.scanId}`);
+        this.logger.setCustomProperties({
+            scanId: onDemandNotificationRequestMessage.scanId,
+            batchJobId: this.currentProcess.env.AZ_BATCH_JOB_ID,
+        });
+
+        const pageScanResult = await this.enqueueNotificationWithRetry(onDemandNotificationRequestMessage);
+
+        this.logger.logInfo(`Writing page notification status to a storage.`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+    }
+
+    private async enqueueNotificationWithRetry(
+        notificationSenderConfigData: OnDemandNotificationRequestMessage,
+    ): Promise<Partial<OnDemandPageScanResult>> {
+        let numberOfTries = 1;
```

Comment:
can we check if `StorageRetryPolicy` w/ exponential retry be used instead https://docs.microsoft.com/en-us/javascript/api/@azure/storage-queue/storageretrypolicy?view=azure-node-latest

Reply:
I don't think we can right now, to use it we need to use 'azure-storage' instead of '@azure/storage-queue' to create the queue service: https://docs.microsoft.com/en-us/azure/storage/queues/storage-nodejs-how-to-use-queues
also we might need to update '@azure/storage-queue' from 10.* to 12.* which is a breaking change we can do in another pr